aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.kt

```diff
@@ -708,15 +775,82 @@ class AWSCognitoAuthPlugin : AuthPlugin<AWSCognitoAuthServiceBehavior>() {
         onSuccess: Consumer<AuthUpdateAttributeResult>,
         onError: Consumer<AuthException>
     ) {
-        TODO("Not yet implemented")
+        authStateMachine.getCurrentState { authState ->
+            when (authState.authNState) {
+                // Check if user signed in
+                is AuthenticationState.SignedIn -> {
+                    GlobalScope.launch {
+                        try {
+                            var accessToken = getAccessToken()
+                            var userAttributes = listOf(
+                                AttributeType.invoke {
+                                    name = attribute.key.keyString
+                                    value = attribute.value
+                                }
+                            )
+                            val userAttributesOptions = options as? AWSCognitoAuthUpdateUserAttributesOptions
+
+                            val userAttributesRequest = UpdateUserAttributesRequest.invoke {
+                                this.accessToken = accessToken
+                                this.userAttributes = userAttributes
+                                this.clientMetadata = userAttributesOptions?.metadata
+                            }
+                            val userAttributeResponse = configureCognitoClients()
+                                .cognitoIdentityProviderClient?.updateUserAttributes(
+                                    userAttributesRequest
+                                )
+                            val result = getUpdateUserAttributeResult(userAttributeResponse, userAttributes)
+                            result[attribute.key]?.let { onSuccess.accept(it) } ?: onError.accept(
+                                AuthException.UserAttributeNotUpdatedException()
+                            )
+                        } catch (e: Exception) {
+                            onError.accept(AuthException.UnknownException(e))
+                        }
+                    }
+                }
+                else -> onError.accept(AuthException.InvalidStateException())
+            }
+        }
     }
 
     override fun updateUserAttribute(
         attribute: AuthUserAttribute,
         onSuccess: Consumer<AuthUpdateAttributeResult>,
         onError: Consumer<AuthException>
     ) {
-        TODO("Not yet implemented")
+        authStateMachine.getCurrentState { authState ->
```

Comment:
This API should call `updateUserAttribute` with default options instead of duplicating the implementation.